centreon-certified/prometheus/prometheus-pushgateway-events-apiv2.lua

```diff
@@ -0,0 +1,378 @@
+  data = data .. '# TYPE ' .. name .. ' counter\n'
+  data = data .. '# HELP ' .. name .. ' 0 is OK, 1 is WARNING, 2 is CRITICAL, 3 is UNKNOWN\n'
+  if not payload.hostgroupsLabel then
+    data = data .. name .. '{label="monitoring_status", host="' .. payload.prom_hname .. '", service="' .. payload.prom_sdesc .. '"} ' .. payload.state .. '\n'
+  else
+    data = data .. name .. '{label="monitoring_status", host="' .. payload.prom_hname .. '", service="' .. payload.prom_sdesc .. '", ' ..  payload.hostgroupsLabel .. '} ' .. payload.state .. '\n'
+  end
+
+
+  local httpRequest = curl.easy()
+  :setopt_url(self.sc_params.params.prometheus_url .. '/metrics/job/' .. self.sc_params.params.prometheus_gateway_job .. '/instance/' .. payload.prom_hname .. '/service@base64/' .. payload.prom_sdesc_url)
+  :setopt_writefunction(
+    function (response)
+      httpResponseBody = httpResponseBody .. tostring(response)
+    end
+  )
+  :setopt(curl.OPT_TIMEOUT, self.sc_params.params.http_timeout)
+  :setopt(
+    curl.OPT_HTTPHEADER,
+    {
+      "content-type: application/openmetrics-text"
+    }
+  )
+
+  -- set proxy address configuration
+  if (self.sc_params.params.proxy_address and self.sc_params.params.proxy_address ~= '') then
+    if (self.sc_params.params.proxy_port and self.sc_params.params.proxy_port ~= '') then
+      httpRequest:setopt(curl.OPT_PROXY, self.sc_params.params.proxy_address .. ':' .. self.sc_params.params.proxy_port)
+    else
+      self.sc_logger:error("EventQueue:send_data: proxy_port parameter is not set but proxy_address is used")
+    end
+  end
+
+  -- set proxy user configuration
+  if (self.sc_params.params.proxy_username ~= '') then
+    if (self.sc_params.params.proxy_password ~= '') then
+      httpRequest:setopt(curl.OPT_PROXYUSERPWD, self.sc_params.params.proxy_username .. ':' .. self.sc_params.params.proxy_password)
+    else
+      self.sc_logger:error("EventQueue:send_data: proxy_password parameter is not set but proxy_username is used")
+    end
+  end
+
+  -- adding the HTTP POST data
+  self.sc_logger:debug("EventQueue:send_data: POST data: '" .. data .. "'")
+  httpRequest:setopt_postfields(data)
+
+  -- performing the HTTP request
+  httpRequest:perform()
+
+  -- collecting results
+  local httpResponseCode = httpRequest:getinfo(curl.INFO_RESPONSE_CODE)
+
+  httpRequest:close()
+
+  -- Handling the return code
+  local retval = false
+  if httpResponseCode == 200 then
+    self.sc_logger:info("EventQueue:send_data: HTTP POST request successful: return code is " .. httpResponseCode)
+    -- now that the data has been sent, we empty the events array
+    self.events = {}
+    retval = true
+  else
+    self.sc_logger:error("EventQueue:send_data: HTTP POST request FAILED, return code is " .. httpResponseCode .. " message is:\n\"" .. httpResponseBody .. "\n\"\n")
+    self.sc_logger:error("the body request " .. data)
+  end
+
+  -- and update the timestamp
+  self.__internal_ts_last_flush = os.time()
```

Comment:
```suggestion
```

this is done by the sc_flush library already https://github.com/centreon/centreon-stream-connector-scripts/blob/8278b591981357b9f17d798c4a8ac77ec6d83961/modules/centreon-stream-connectors-lib/sc_flush.lua#L107